api/prisma/schema.prisma

```diff
@@ -145,6 +145,145 @@ model user {
 
 // -----------------------------------
 
+/// An exam as designed by the examiners
+model NewExam {
+  /// Globally unique exam id
+  ///
+  /// NOTE: Field is not "default", because of when/where exams are created.
+  id                 String         @id @map("_id") @db.ObjectId
+  /// Human-readable exam name
+  name               String
+  /// Notes given about exam
+  accessibility_note String
+  /// All questions for a given exam
+  question_types     QuestionType[]
+  /// Configuration for exam metadata
+  config             NewConfig
+}
+
+type QuestionType {
+  /// Unique question type id
+  id        String          @db.ObjectId
+  type      NewQuestionType
+  text      String?
+  questions NewQuestion[]
+}
+
+/// A multiple choice question
+type NewQuestion {
+  /// Unique question id
+  id         String      @db.ObjectId
+  /// Main question paragraph
+  text       String
+  /// Zero or more tags given to categorize a question
+  tags       String[]
+  /// Optional audio for a question
+  audio      NewAudio?
+  /// Available possible answers for an exam
+  answers    NewAnswer[]
+  /// TODO Possible "deprecated_time" to remove after all exams could possibly have been taken
+  deprecated Boolean
+}
+
+type NewAudio {
+  /// Optional text for audio
+  captions   String?
+  /// URL to audio file
+  url        String
+  /// Start time of audio clip for question
+  ///
+  /// If not set, audio starts at 0
+  start_time Float?
+  /// End time of audio clip for question
+  ///
+  /// If not set, audio ends at the end of the clip
+  end_time   Float?
+}
+
+enum NewQuestionType {
+  /// Single question with one or more answers
+  MultipleChoice
+  /// Mass text
+  Dialogue
+}
+
+type NewAnswer {
+  id         String  @db.ObjectId
+  is_correct Boolean
+  text       String
+}
+
+type NewConfig {
+  tags           NewTagConfig[]
+  /// Total time allocated for exam in seconds
+  total_time     Int
+  question_types QuestionConfig[]
+}
+
+type QuestionConfig {
+  type                        NewQuestionType
+  number_of_type              Int
+  number_of_questions         Int
+  number_of_correct_answers   Int
+  number_of_incorrect_answers Int
+}
+
+type NewTagConfig {
+  /// Foreign key to tags
+  set                 String[] @db.ObjectId
```

Comment:
Should this be a relation?

Reply:
I am not sure how to write that relation?

`exam.question_types[].tags` can be a subset of `exam.config.tags[].set`. Or, `exam.config.tags[].set` can be a subset of `exam.question_types[].tags`.

So, all of `exam.config.tags[].set` must be in the union of `exam.question_types.tags`.